src/main/java/org/terasology/logic/health/RegenComponent.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.logic.health;
+
+import org.terasology.entitySystem.Component;
+import org.terasology.math.TeraMath;
+import org.terasology.network.Replicate;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class RegenComponent implements Component {
+    private Map<String, Float> regenValue = new HashMap<>();
+    private Map<String, Long> regenEndTime = new HashMap<>();
+    public long lowestEndTime;
```

Comment:
I wonder if the wording could be improved from "lowest" here - to me at least it doesn't seem super intuitive vs the concepts of starting and ending things. `soonest` instead perhaps?